python/ray/rllib/agents/mock.py

```diff
@@ -32,11 +34,15 @@ def _train(self):
         if self.config["mock_error"] and self.iteration == 1 \
                 and (self.config["persistent_error"] or not self.restored):
             raise Exception("mock error")
-        return dict(
+        result = dict(
             episode_reward_mean=10,
             episode_len_mean=10,
             timesteps_this_iter=10,
             info={})
+        if self.config["user_checkpoint_freq"] > 0 and self.iteration > 0:
+            if self.iteration % self.config["user_checkpoint_freq"] == 0:
+                result.update({tune_result.SHOULD_CHECKPOINT: True})
```

Comment:
What if user has explicitly said `tune_result.SHOULD_CHECKPOINT: False`? Should this be 
```suggestion
                result.setdefault(tune_result.SHOULD_CHECKPOINT, True)
```

Reply:
Oh, this is only for testing purposes and isn't extended.

Reply:
Lol fo course, my bad